src/js/intercept-category-navigation.js

```diff
@@ -0,0 +1,50 @@
+const getQuestions = async () => {
+  const response = await fetch('/questions.json');
+  const questions = await response.json();
+
+  return questions;
+};
+
+const getRelevantQuestions = (questions, category, isMultipleChoice) => {
+  return questions.filter((question) => {
+    if (!isMultipleChoice && question['Multiple Choice Only']) {
+      return false;
+    }
+
+    if (category !== 'All' && !question.Tags.some((tag) => tag === category)) {
+      return false;
+    }
+
+    return true;
+  });
+};
+
+(async () => {
+  sessionStorage.removeItem('questions');
```

Comment:
It appears that there are two immediately invoked functions that are accessing `sessionStorage`. I think we are ok, but that makes me think there is a possibility that since one of these functions is async we might run into where the reads or writes to sessionStorage conflict. Thoughts?

Reply:
`intercept-category-navigation.js` only gets loaded on the home page, and `update-pagination-links.js` only gets loaded on the individual question pages, so they shouldn't ever cross paths. Instead of bundling all the JS into one script that gets served on every page, which is weirdly the norm for a lot of sites, I opted to only give each page as much JS as they need.